Healthy/Classes/ReusableViews/CheckboxButton.swift

```diff
@@ -0,0 +1,30 @@
+import UIKit
+
+class CheckboxButton: UIButton {
+
+    // Images
+    let checkedImage = UIImage.icCheckboxSelected
+    let uncheckedImage = UIImage.icCheckboxNotSelected
+
+    // Bool property
+    var isChecked: Bool = false {
+        didSet {
+            if isChecked == true {
+                self.setImage(checkedImage, for: UIControl.State.normal)
+            } else {
+                self.setImage(uncheckedImage, for: UIControl.State.normal)
+            }
+        }
+    }
+
+    override func awakeFromNib() {
+        self.addTarget(self, action: #selector(buttonClicked(sender:)), for: .touchUpInside)
+        self.isChecked = false
+    }
+
+    @objc func buttonClicked(sender: UIButton) {
+        if sender == self {
+            isChecked = !isChecked
+        }
+    }
```

Comment:
@objc private func buttonClicked(sender: UIButton) {
            isChecked = !isChecked
    }
if you set the method to private you don't need to check the sender

Reply:
Done !  👍🏼